sympy/physics/quantum/sho1d.py

```diff
@@ -454,8 +454,8 @@ class Hamiltonian(SHOOp):
 
         >>> H = Hamiltonian('H')
         >>> k = SHOKet('k')
-        >>> qapply(H*k)
-        hbar*k*omega*|k> + hbar*omega*|k>/2
+        >>> qapply(H*k, mul=False)
+        hbar*omega*(k + 1/2)*|k>
```

Comment:
I get an error testing this locally:
```console
$ cat tq.py 
from sympy.physics.quantum import qapply
from sympy.physics.quantum.sho1d import Hamiltonian, SHOKet
H = Hamiltonian('H')
k = SHOKet('k')
print(qapply(H*k, mul=False))
$ python tq.py 
Traceback (most recent call last):
  File "tq.py", line 5, in <module>
    print(qapply(H*k, mul=False))
  File "/home/oscar/current/sympy/sympy.git/sympy/physics/quantum/qapply.py", line 740, in qapply
    res = qapply_Mul2([e], [])
  File "/home/oscar/current/sympy/sympy.git/sympy/physics/quantum/qapply.py", line 293, in qapply_Mul2
    res0 = qapply_Mul2Atoms(lhsR, rhsL)  # this may return ANY type!
  File "/home/oscar/current/sympy/sympy.git/sympy/physics/quantum/qapply.py", line 434, in qapply_Mul2Atoms
    result = cast(Any, lhs)._apply_operator(rhs, **options)
  File "/home/oscar/current/sympy/sympy.git/sympy/physics/quantum/operator.py", line 169, in _apply_operator
    return dispatch_method(self, '_apply_operator', ket, **options)
  File "/home/oscar/current/sympy/sympy.git/sympy/physics/quantum/qexpr.py", line 407, in dispatch_method
    result = f(arg, **options)
TypeError: _apply_operator_SHOKet() got an unexpected keyword argument 'mul'
```

Reply:
It seems a lot of the examples don't work locally and fail with the same error. I'm not sure how it is that the tests have passed but yet this fails on my machine.

Reply:
It's clear that the method does not have the keyword argument:
```console
$ git grep _apply_operator_SHOKet
sympy/physics/quantum/sho1d.py:    def _apply_operator_SHOKet(self, ket):
sympy/physics/quantum/sho1d.py:    def _apply_operator_SHOKet(self, ket):
sympy/physics/quantum/sho1d.py:    def _apply_operator_SHOKet(self, ket):
sympy/physics/quantum/sho1d.py:    def _apply_operator_SHOKet(self, ket):
```

Reply:
This has been fixed with PR #24404. It seems that is not active in your local environment. For qapply to work all PRs I have put in in the last two months are required.

Reply:
Oh, okay so does this PR need to be rebased or merged with current master then? When the tests run in CI it merges the PR branch into a copy of master first.

Reply:
Actually I believed this PR is based on the current master (but I have to admit that I know little about git, so that is probably not correct). Fact is the previous PRs I have done make sense in itself and fix issues with the original quantum package. The new qapply relies on most of these patches to be in place. 
So what needs to be done to incorporate this dependency?

Reply:
This PR will have been based on master from before #24404 was merged. So the PR branch itself does not include the changes from #24404. The tests pass in CI because they don't run directly on the PR branch but rather on a temporary branch that is the result of merging the PR into a copy of the master branch. That's why the test results I see locally are different from those in CI.

If you check out the `qapply-rewrite` branch yourself do you not see the same error I showed above?

What needs to be done is to either merge master into this branch or rebase this branch onto master. I would be inclined to rebase and also squash all the commits at the same time because I don't think that these 54 separate commits are really needed as separate commits:
https://github.com/sympy/sympy/pull/24349/commits

Whatever you do make sure to save a backup copy of this branch and perhaps also a backup copy of the files that have been changed here just in case something goes wrong with the merge/rebase. People who are relatively new to git can often end up accidentally deleting all their work when doing something like a merge or rebase. (It wouldn't be the first time I've explicitly said to make a backup and then someone overlooked that point!)

Reply:
Will do it in the next days when code base is fixed, as I haven't done it before.

Reply:
Bear with me as a novice user of gib. I'm managing git using the web UI of github, so I have limited options and it seems I cannot do a rebase, but only merge sympy:master into my branch.
I have thus synced my fork of sympy:master, and synced my branch qapply-rewrite from that fork. As a result my branch is now reported as "62 commits ahead of sympy:master" and my code base now incorporates  #24404 and all other PRs.  So this is the merge of sympy:master into my branch. 
I agree that the commits could be squashed, however don't see how to accomplish that from the web UI.

Reply:
I don't recommend trying to do everything through the web UI.